pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -649,6 +652,18 @@ func (c *Controller) Run(stop <-chan struct{}) {
 	go c.exports.Run(stop)
 	kubelib.WaitForCacheSync("kube controller", stop, c.informersSynced)
 	log.Infof("kube controller for %s synced after %v", c.opts.ClusterID, time.Since(st))
+	if c.ambientIndex != nil {
```

Comment:
Is there a way to describe this behavior with a unit test in `controller_test.go`?

Reply:
Added.. its not pretty but maybe good enough